core/src/main/java/org/mskcc/cbio/oncokb/util/SummaryUtils.java

```diff
@@ -476,28 +520,141 @@ private static String enrichGeneEvidenceDescription(EvidenceType evidenceType, G
+    }
 
-        if (highestOncogenicity != null && (highestOncogenicity.equals(Oncogenicity.YES) || highestOncogenicity.equals(Oncogenicity.LIKELY))) {
+    public static Oncogenicity getRefAlleleOncogenicityBasedOnAltAlleles(List<Oncogenicity> oncogenicities) {
+        if (oncogenicities.size() == 0) {
+            return Oncogenicity.UNKNOWN;
+        } else if (oncogenicities.size() == 1) {
+            if (Arrays.asList(Oncogenicity.YES, Oncogenicity.LIKELY).contains(oncogenicities.get(0))) {
+                return Oncogenicity.LIKELY;
+            } else {
+                return Oncogenicity.UNKNOWN;
+            }
+        } else {
+            if (oncogenicities.stream().anyMatch(oncogenicity -> Arrays.asList(Oncogenicity.YES, Oncogenicity.LIKELY).contains(oncogenicity))) {
+                return Oncogenicity.LIKELY;
+            } else {
+                return Oncogenicity.UNKNOWN;
+            }
+        }
+    }
+
+    public static String alleleSummary(ReferenceGenome referenceGenome, Alteration alteration, String queryHugoSymbol, boolean isVus) {
+        StringBuilder sb = new StringBuilder();
+
+        Set<Alteration> alleles = new HashSet<>(AlterationUtils.getAlleleAlterations(referenceGenome, alteration));
+
+        Map<Oncogenicity, Set<Alteration>> map = geAlterationsWithHighestOncogenicity(new HashSet<>(alleles));
+        List<Oncogenicity> allowedOncogenicity = Arrays.asList(Oncogenicity.YES, Oncogenicity.LIKELY, Oncogenicity.LIKELY_NEUTRAL, Oncogenicity.RESISTANCE);
+        List<Oncogenicity> allelesOncogenicities = allowedOncogenicity
+            .stream()
+            .filter(oncogenicity -> map.containsKey(oncogenicity))
+            .collect(Collectors.toList());
+        boolean hasOncogenic = allelesOncogenicities
+            .stream()
+            .filter(oncogenicity -> Arrays.asList(Oncogenicity.YES, Oncogenicity.LIKELY).contains(oncogenicity))
+            .count() > 0;
+        List<String> oncogenicitySentences = allelesOncogenicities
+            .stream()
+            .map(oncogenicity -> alleleSummaryWithOncogenicities(alteration.getGene().getHugoSymbol(), oncogenicity, map.get(oncogenicity)))
+            .collect(Collectors.toList());
+
+        if (oncogenicitySentences.size() > 0) {
             String altStr = getGeneMutationNameInVariantSummary(alteration.getGene(), referenceGenome, queryHugoSymbol, alteration.getAlteration());
-            sb.append("The " + altStr + " has not been functionally or clinically validated.");
+            Oncogenicity referredOncogenicity = getRefAlleleOncogenicityBasedOnAltAlleles(allelesOncogenicities);
+            if (referredOncogenicity == null) {
+                referredOncogenicity = Oncogenicity.UNKNOWN;
+            }
+            if (isVus) {
+                sb.append(getVUSSummary(alteration, altStr, true));
+            } else {
+                sb.append("The " + altStr + " has not specifically been reviewed by the OncoKB team.");
```

Comment:
Is the word `mutation` after the `altStr` not needed anymore? I see in the doc that some places don't have it, but others do.

Reply:
`altStr` includes `mutation` when necessary.